File: C++/Q13.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main() {
    float a, b, c, discriminant, root1, root2;
    cout << "Enter coefficients a, b, c: ";
    cin >> a >> b >> c;
    discriminant = b * b - 4 * a * c;

    if (discriminant > 0) {
        root1 = (-b + sqrt(discriminant)) / (2 * a);
        root2 = (-b - sqrt(discriminant)) / (2 * a);
        cout << "Roots are real and different: " << root1 << ", " << root2;
    } else if (discriminant == 0) {
        root1 = -b / (2 * a);
        cout << "Roots are real and same: " << root1;
    } else {
        float real = -b / (2 * a);
        float imag = sqrt(-discriminant) / (2 * a);
        cout << "Complex Roots: " << real << " ± " << imag << "i";
    }
    return 0;
}

File: Python/Q13.py
# Q13.py - Program to Find All Roots of a Quadratic Equation
import math
a = float(input("Enter a: "))
b = float(input("Enter b: "))
c = float(input("Enter c: "))
d = b**2 - 4*a*c
if d > 0:
    root1 = (-b + math.sqrt(d)) / (2*a)
    root2 = (-b - math.sqrt(d)) / (2*a)
    print(f"Roots are real and different: {root1}, {root2}")
elif d == 0:
    root = -b / (2*a)
    print(f"Roots are real and same: {root}")
else:
    real = -b / (2*a)
    imag = math.sqrt(-d) / (2*a)
    print(f"Roots are complex: {real}+{imag}i, {real}-{imag}i")